_pytest/skipping.py

```diff
@@ -230,7 +230,8 @@ def pytest_runtest_makereport(item, call):
     if hasattr(item, '_unexpectedsuccess') and rep.when == "call":
         # we need to translate into how pytest encodes xpass
         rep.wasxfail = "reason: " + repr(item._unexpectedsuccess)
-        rep.outcome = "failed"
+        # TODO: Do we need to check for strict xfail here as well?
```

Comment:
Oh OK, I looked some moreand noticed that in this case it is just to cover `unittest` specifically, which doesn't have a strict option:

```python
import unittest

class T(unittest.TestCase):

    @unittest.expectedFailure 
    def test_foo(self):
        assert 0

    @unittest.expectedFailure 
    def test_pass(self):
        assert 1

if __name__ == '__main__':
    unittest.main()        
```

Running it:

```
$ python foo.py
xu
----------------------------------------------------------------------
Ran 2 tests in 0.000s

FAILED (expected failures=1, unexpected successes=1)
```

So I think the original code was the correct one.